sdk/go/pulumi-language-go/main.go

```diff
@@ -1087,7 +1087,7 @@ func (host *goLanguageHost) InstallDependencies(
 	cmd.Stdout, cmd.Stderr = stdout, stderr
 
 	if err := cmd.Run(); err != nil {
-		return fmt.Errorf("`go mod tidy` failed to install dependencies: %w", err)
+		return fmt.Errorf("`go mod tidy` failed to install dependencies for %s: %w", req.Info.ProgramDirectory, err)
```

Comment:
I'm not sure we want this in the error generally? Normally users only have one program directory and they know what it is so this just makes the error longer.